Add pfcwd utility to the config/show CLI

andriymoroz-mlnx opened the pull request:
Signed-off-by: Andriy Moroz <c_andriym@mellanox.com>

<!--
Please make sure you've read and understood our contributing guidelines:
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "closes #xxxx",
"fixes #xxxx" or "resolves #xxxx" so that GitHub automatically closes the related
issue when the PR is merged

If you are modifying or removing an existing 'show', 'config' or 'sonic-clear'
subcommand, or you are adding a new subcommand, please make sure you also
update the Command Line Reference Guide (doc/Command-Reference.md) to reflect
your changes.

Please provide the following information:
-->

**- What I did**

**- How I did it**

**- How to verify it**

**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**

-->

andriymoroz-mlnx committed:
Add pfcwd utility to the config/show CLI

Signed-off-by: Andriy Moroz <c_andriym@mellanox.com>

andriymoroz-mlnx committed:
Add some more commands

Signed-off-by: Andriy Moroz <c_andriym@mellanox.com>

tsvanduyn committed:
[config] Add commands for adding/removing NTP servers (#611)

stcheng committed:
[neighbor_advertiser]: Add V6 support to mirror ICMPV6 packets (#612)

Add two entries to mirror ICMPV6 type 128 and 129 packets.

Signed-off-by: Shu0T1an ChenG <shuche@microsoft.com>

yxieca committed:
[neighbor advertiser] try getting vlan addresses from o.s first (#613)

* [neighbor advertiser] try getting vlan addresses from o.s first

The IPv6 link local address can only be obtained from the o.s., not the
config_db.

Anything not obtained from o.s. will be retrieved from config_db.

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

* address review issues

* Update neighbor_advertiser

tsvanduyn committed:
[show] Add command 'show runningconfiguration syslog' (#617)

tsvanduyn committed:
[show] Add command 'show runningconfig ntp' (#620)

tsvanduyn committed:
Added check to config syslog del to verify if ip address was configured. (#622)

daall committed:
[fast-reboot] Check if ASIC config has changed before warm reboot (#621)

[fast-reboot] Check if ASIC config has changed before fast reboot
* Adds script to compare checksums for config files between SONiC images
* Adds pre-check to fast reboot script to confirm config files match before performing fast reboot

Signed-off-by: Danny Allen <daall@microsoft.com>

* Fix formatting

* Incorporate feedback

* Add check for reboot type and override ability

* Clean up error checking

stcheng committed:
[neighbor_advertiser]: Change the ICMPv6 type to 135 (#629)

Need to mirror the neighbor solicitation packets - type 135.

Signed-off-by: Shu0T1an ChenG <shuche@microsoft.com>

tsvanduyn committed:
[show] Add 'show boot' command (#630)

yxieca committed:
[warm/fast reboot] provide strict option to prevent warm reboot under certain conditions (#631)

- the only condition now is lack of control plane assistant.

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

kannankvs committed:
updated "show ntp" to use cgexec when management vrf is enabled (#627)

pavel-shirshov committed:
[FastReboot]: Send SIGINT to all teamd before stop (#633)

* [FastReboot]: Send SIGINT to all teamd before stop

* exit code == 1 is valid for pkill

* Check that all teamd are stopped before to kill the container

tsvanduyn committed:
[config] Fix naming conflict between 'config syslog' function and 'syslog' module (#636)

daall committed:
[warm reboot] Skip ASIC config pre-check if current image does not support it (#637)

Signed-off-by: Danny Allen <daall@microsoft.com>

rvisnu committed:
[config] Add commands to remove BGP neighbor configuration (#625)

liat-grozovik commented on config/main.py:
```diff
@@ -614,6 +614,91 @@ def remove(session_name):
     config_db.connect()
     config_db.set_entry("MIRROR_SESSION", session_name, None)
 
+#
```
do we also have a show command that should be added as well?

andriymoroz-mlnx replied:
yes, show is also updated in this PR

liat-grozovik commented on doc/Command-Reference.md:
```diff
@@ -2509,6 +2510,65 @@ This command displays a list of NTP peers known to the server as well as a summa
 Go Back To [Beginning of the document](#SONiC-COMMAND-LINE-INTERFACE-GUIDE) or [Beginning of this section](#NTP)
 
+# PFC Watchdog Commands
+
```
i would suggest to have a high level description of the feature before the commands description just to be clear how it works and how the configuration affect the feature behaviour. should not be long but clear enough

in addition each parameter has a default configuration, it is highly recommended to provide the default. In case it is platform dependent worth mentioning it.

andriymoroz-mlnx replied:
Added wiki reference and default parameters description

jleveque committed:
[config] Reset failed status of all SONiC services, whether or not they are currently failed (#619)

stcheng committed:
[warm-reboot]: Do not clean up mirror session state database (#639)

In order to maintain the same monitor port across the warm reboot,
exclude the mirror session state database.

Signed-off-by: Shu0T1an ChenG <shuche@microsoft.com>

andriymoroz-mlnx committed:
Add SSD Health CLI utility (#587)

* Add SSD Health CLI utility

Signed-off-by: Andriy Moroz <c_andriym@mellanox.com>

* Add SSD Health command reference

* Rename ssdutility entrypoint to ssdutil

Signed-off-by: Andriy Moroz <c_andriym@mellanox.com>

* Fix - do not add units if value is N/A

Signed-off-by: Andriy Moroz <c_andriym@mellanox.com>

nazariig committed:
Fixed config Asym PFC CLI. (#632)

Signed-off-by: Nazarii Hnydyn <nazariig@mellanox.com>

yxieca committed:
Revert "[FastReboot]: Send SIGINT to all teamd before stop (#633)" (#650)

This reverts commit 7cf821c440703f55af1c125092272a1ff69ce752.

jleveque committed:
[config] Add '-h' and '-?' as help options to unify with show/sonic-clear (#647)

pavel-shirshov committed:
Use SIGUSR2 to stop teamd before fast-reboot (#660)

yxieca committed:
[warm/fast reboot] some service docker might have been stopped already (#668)

* [warm/fast reboot] some service docker might have been stopped already

Some services are depending on swss and might have been stopped at the
time the loop of docker killing was executed.

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

* log and ignore all errors while killing containers

* add debug message

* put redirect at the right place

stephenxs committed:
[Mellanox] Don't start & stop pmon when config reload (#671)

stcheng committed:
[acl_loader]: Add monitor port column in show mirror_session output (#662)

This column helps checking the output monitor port for the mirror
session.

Signed-off-by: Shu0T1an ChenG <shuche@microsoft.com>

daall committed:
[neighbor_advertiser] Verify that DIPs returned from ferret are not in device VLAN (#670)

* [neighbor_advertiser] Verify that DIPs returned from ferret are not in device VLANs
* Add a method in neighbor_advertiser to check if a DIP is in any of the device's VLANs
* Check that DIPs returned from ferret are not in any of the device's VLANs before proceeding
* Add a retry to attempt to request different DIPs from a given ferret VIP

Signed-off-by: Danny Allen daall@microsoft.com

* Remove extraneous local var

* Handle error cases

* Incorporate PR feedback

* Revert structure and clarifying comments

* Show retry attempts

* Clarify device vlan check

* Filters out interfaces that have no subnet or an IPv6 subnet

* Refactor to use is_ip_prefix_in_key

* Fix runtime errors

yxieca committed:
[fast reboot] kill teamd docker directly (#683)

Letting teamd quit and send out last handshake cause some unexpected
interaction with orchagent. It appears that orchagent noticed these lag
were removed and start to tear down the LAG on the ASIC side as well.
This would cause the control plane to go down immediately.

Until we have a better solution, revert back to original behavior.

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

jleveque committed:
[show] Properly replace port name with alias in command output (#664)

AmitKaushik7 committed:
[pfcwd] Apply pfcwd config only on physical ports (#640)

wendani committed:
show pfcwd status to be 'N/A' when pfcwd is stopped (#682)

Signed-off-by: Wenda Ni <wenni@microsoft.com>

wendani committed:
Add sub port interface to `config interface ip add/del` (#651)

* Support sub port interface admin status change

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Initialize vlan_id as an empty string

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Remove parentheses

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Take care of parent port being port channel

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Add sub port interface to `config interface ip add/del`

Signed-off-by: Wenda Ni <wenni@microsoft.com>

wendani committed:
Support sub port interface admin status change (#638)

* Support sub port interface admin status change

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Initialize vlan_id as an empty string

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Remove parentheses

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Take care of parent port being port channel

Signed-off-by: Wenda Ni <wenni@microsoft.com>

yxieca committed:
Revert "[fast reboot] kill teamd docker directly (#683)" (#690)

This reverts commit 592413b274c371ba284a159bc7723192f0b9491c.

neethajohn committed:
Address the 'show ipv6 interface' crash and include the missed bgp columns (#686)

* Address the 'show ipv6 interface' command crash

Signed-off-by: Neetha John <nejo@microsoft.com>

* Include the bgp info that was missed in "show ipv6 interface" output

Signed-off-by: Neetha John <nejo@microsoft.com>

samaity committed:
[config]: Add hostname command to config CLI (#669)

Signed-off-by: Sangita Maity <sangitamaity0211@gmail.com>

pavel-shirshov committed:
Fix bgp 'show' and 'clear' commands (#692)

* Fix bgp commands

* Make the function more pythonic

* Fix clear command for current frr

Commit:
[sonic-utilities] managementVRF cli support(l3mdev) (#463)

* [sonic-utilities] managementVRF cli support(l3mdev)

stepanblyschak committed:
[acl-loader] egress mirror action support and action ASIC support check (#575)

- support egress mirror action
- support action check via state DB

mykolaf committed:
[intfstat,portstat] fix table_as_json (#691)

Signed-off-by: Mykola Faryma <mykolaf@mellanox.com>

stcheng committed:
[neighbor_advertiser]: Adapt to different mirror ACL table names (#703)

ACL table names varies on different platforms.
The current assumption is that the name would contain either
EVERFLOW or EVERFLOWV6.

Thus a function find_mirror_table_name is added to find the exact
names for both the v4 and v6 mirror table names.

Signed-off-by: Shu0T1an ChenG <shuche@microsoft.com>

lyndonsiao committed:
[netstat]: Fix for negative output values of counters after clear counters operation (#697)

Immediately after a clear counter operation, the difference between new
counter and old counter is negative. Returning 0 in this situation

Kalimuthu-Velappan committed:
[generate_dump] Make regex more specific for disabling/enabling logrotate (#701)

jleveque committed:
[Command Reference] Unify style and formatting; Fix organization; Other fixes (#707)

- Unify style and Markdown formatting
- Fix organization of recently-added commands
- Modify heading sizes
- Fix spelling, grammar and other errors
- Fix bad links
- Add 'sonic_installer cleanup' command
- Add 'sudo' to sonic_installer command examples

jleveque committed:
[PR template] Add reminder to add/modify/remove unit tests as appropriate (#708)

yxieca committed:
[command reference] add warm reboot command document (#704)

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

yxieca committed:
[warm/fast reboot] continue executing when killing docker failed (#713)

* [warm/fast reboot] continue executing when killing docker failed

After killing bgpd, the execution has passed point of safe return. We
need to proceed and ignore any trivial failures.

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

* mute stderr

stcheng committed:
[neighbor_advertiser]: Add sleep in setting mirror session and ACL rules (#714)

The sleep(3) is to ensure that orchagent could execute tasks by the
correct sequence so that mirror session and ACL rules could be
successfully created/removed.

Signed-off-by: Shu0T1an ChenG <shuche@microsoft.com>

andriymoroz-mlnx committed:
[doc] Add start_default description

Signed-off-by: Andriy Moroz <c_andriym@mellanox.com>

stcheng committed:
[config]: Flush the neighbor table when removing the router interface (#606)

Manually trigger the flush command to remove the deprecated
neighbors that are associated with the removed router interface.

Signed-off-by: Shu0T1an ChenG <shuche@microsoft.com>

renukamanavalan committed:
Add a generic configlet application script (#716)

* A generic configlet application script.

* Per review comments by Joe.

Commit:
Management vrf snmp cli support (#472)

* [sonic-utilities] managementVRF cli support(l3mdev)

This commit adds CLI support for management VRF using l3dev. mVRF can
be enabled using config vrf add mgmt and deleted using config vrf del mgmt.
Show commands for management VRF are added which displays the linux command
output, will update show command display after concluding what would be the
output for the show commands.
Added cli to configure management interface(eth0), config interface ip eth0
add can be used to configure eth0 ip and config ip eth0 remove is used to
remove eth0 ip.

New cli config/show commands:

config vrf add mgmt
config vrf del mgmt
config interface eth0 ip add ip/mask gatewayIP
config interface eth0 ip remove ip/mask

show mgmt-vrf
show mgmt-vrf route
show mgmt-vrf addresses
show mgmt-vrf interfaces

Signed-off-by: Harish Venkatraman <harish_venkatraman@dell.com>

wendani committed:
show subinterfaces status (#642)

* Add sub port interface to intfutil status

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Format change

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Hook 'intfutil status' with show subinterfaces status

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Add 'intfutil status subport' to show only the status of sub port interfaces

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Hook 'intfutil status subport' with 'show subinterfaces status'

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Support intfutil status <sub_port_intf_name>

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Add alias naming mode support

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Address comments

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Address comment: concatenate command line, and validate input

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Address comment: keep show interface status intact

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Add unit test for show interfaces status / intfutil status

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Add unit test for 'show interfaces status --verbose'

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Add unit test for 'show subinterfaces status' / 'intfutil status subport'

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Add unit test for 'show subinterfaces status --verbose'

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Add unit test for single sub interface status show

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Add unit test for '--verbose' single sub interface status show

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Add unit test for showing status of single sub interface in alias naming mode

Signed-off-by: Wenda Ni <wenni@microsoft.com>

* Add unit test for '--verbose' status of single sub interface in alias naming mode

Signed-off-by: Wenda Ni <wenni@microsoft.com>

GarrickHe committed:
sonic-utilities: Add support for sFlow (#592)

* sonic-utilities: Add sFlow CLIs

mykolaf committed:
Revert "show BPS, PPS, UTIL rates w/o previous clear (#508)" (#718)

* Revert "show BPS, PPS, UTIL rates w/o previous clear (#508)"

This reverts commit 148d455c1bf8c8fc9b3e7dee279791866b165b4f.

* [portstat] use actual port speed for util calculation after revert

Signed-off-by: Mykola Faryma <mykolaf@mellanox.com>

renukamanavalan committed:
Make configlet application script idempotent for updates. (#728)

* Make script idempotent, for updates.
Delete of non-existing table/keys is a no-op, hence can be ignored.

* Name change

heidinet2007 committed:
Add an on/off knob for BGP EOIU pulling on warm restart (#655)

neethajohn committed:
Do not start pfcwd for M0 devices (#726)

* Do not start pfcwd for M0 devices

Signed-off-by: Neetha John <nejo@microsoft.com>

* Avoid exiting for device type error

Signed-off-by: Neetha John <nejo@microsoft.com>

prsunny committed:
Adding LOOPBACK interface to db migrate script (#734)

andriymoroz-mlnx committed:
Merge remote-tracking branch 'upstream/master' into pfcwd_connect

wangshengjun committed:
[config]: add the vid range check for creating vlan. (#634)

Signed-off-by: wangshengjun <wangshengjun@asterfusion.com>

lguohan committed:
[bugfix]: allow loopback vrf migration from version 1_0_1 (#737)

commit fe4446a4808bc98ca2b24b0a9afe822aaa50a7ce fix loopback vrf migration.

but the commit does not enable db migration from version 1_0_1.

Now, bump version 1_0_2 and allow db fix from 1_0_1

PrabhuSreenivasan committed:
SONiC Management Framework Release 1.0 (#659)

* Added scripts to provide back-end for CLI commands

Scripts to provide the files /var/platform/syseeprom and /var/paltform/system,
which are JSON files read by CLI commands, to provide system information.

Signed-off-by: Howard Persh <Howard_Persh@dell.com>

* Change system information poll to not use file locking

Signed-off-by: Howard Persh <Howard_Persh@dell.com>

* Modify system system info poller to provide process start time in seconds

Signed-off-by: Howard Persh <Howard_Persh@dell.com>

* Small optimization for system info poll script

Signed-off-by: Howard Persh <Howard_Persh@dell.com>

* Add new scripts for CLI backend to setup.py

Signed-off-by: Howard Persh <Howard_Persh@dell.com>

renukamanavalan committed:
Revert "SONiC Management Framework Release 1.0 (#659)" (#741)

This reverts commit e0e8c2f53a61eeb1544b244f68e88b1cca7abef4.

daall committed:
Add CLI support for configurable drop counters (#688)

* Add CLI support for configurable drop counters
- Adds dropconfig script to configure drop counters
- Adds dropstat script to show and clear drop counters
- Adds handlers for drop counters to show, config, and sonic-clear

Signed-off-by: Danny Allen <daall@microsoft.com>

* Update command naming

* Add safety check for unsupported drop counters/reasons

* Make drop reasons more easily readable

* Respond to community feedback

* Update test to match command

* Add reference for show commands

* Finish command reference

* Fix error messages

* Update command name

* Make subcommands less verbose

* Add safety check for number of counters in use

daall committed:
[neighbor_advertiser] Adds initial support for HTTPS to neighbor advertiser (#750)

* [neighbor_advertiser] Adds initial support for HTTPS to neighbor advertiser

Signed-off-by: Danny Allen <daall@microsoft.com>

* Add debug logs for HTTP failover

* Make debug logs more explicit

pra-moh committed:
[show] Add 'features' subcommand to display status for optional features (#712)

Commit:
[VRF]: submit vrf CLI #392  (#558)

* Support VRF

dzhangalibaba committed:
[multiDB]: all application should use API to get redis_client (#753)

renukamanavalan committed:
Fix a bug in idempotent check. (#755)

Currently it misses the case where data for a new key is empty.

pra-moh committed:
[config] Add 'feature' subcommand (#746)

yxieca committed:
[neighbor advertiser] catch all exceptions while trying https endpoint (#757)

* [neighbor advertiser] catch all exceptions while trying https endpoint

When connecting https endpoint failed, we need to try http endpoint.
Therefore we need to catch all exceptions.

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

* raise exception if https returned non-success status code

rajendra-dendukuri committed:
Use config-setup infrastructure to take backup of SONiC configuration (#715)

Used "config-setup backup" command to take a backup copy of current SONiC configuration.

Refer to Azure/SONiC#433 and Azure/sonic-buildimage#3227 for more information

Added skip_migration option to "sonic_installer install" command to allow user to install an image
with factory default configuration.

Signed-off-by: Rajendra Dendukuri <rajendra.dendukuri@broadcom.com>

yxieca committed:
[neighbor advertiser] raise exception when http endpoint return failure (#758)

* [neighbor advertiser] raise exception when http endpoint return failure

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

* Delete deprecated error checking code

* Add back check for None

santhosh-kt committed:
Platform plugin for Fast-reboot/warm-reboot  (#740)

* Fastboot/warmboot platform plugin

* Added platform plugin

FuzailBrcm committed:
Platform Driver Development Framework (PDDF): Adding PDDF CLI utils (#624)

This change is related to Platform Driver Development Framework (PDDF) which is being added to sonic-buildimage repo. More details can be found here, Azure/SONiC#406

PDDF supports its own CLI utilities, which use generic PDDF component plugins. I added these PDDF CLI utilities.

yxieca committed:
[fast/warm reboot] ignore errors after shutting down critical service(s) (#761)

Once any critical service is shutdown (radv/swss/syncd), we have to
commit to the reboot. Failing in the middle will leave the system in
bad state.

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

mykolaf committed:
[command reference] Update show int counters info with l3 RIF (#725)

Signed-off-by: Mykola Faryma <mykolaf@mellanox.com>

renukamanavalan committed:
A generic JSON file updater, which can add/update-existing attributes. (#770)

* A generic JSON file updater, which can add/update-existing attributes.
This tool would be used to update /etc/sonic/core_analyzer.rc.json file to add
credentials by HW proxy.

* Updated per review comments.
The option is better named.

hpersh committed:
[scripts] Adds support for management (1.0) commands (#745)

Add utility to translate text output of syseeprom dump into JSON, for
consumption by new management (1.0) command.

Signed-off-by: Howard Persh <hpersh@yahoo.com>

mykolaf committed:
[intfstat] fix --delete fail when counters dump does not exist (#752)

Signed-off-by: Mykola Faryma <mykolaf@mellanox.com>

kannankvs committed:
Add VRF-related commands to Command Reference Document (#778)

mykolaf committed:
[portstat] fix header used (#743)

* [portstat] fix header used

Signed-off-by: Mykola Faryma <mykolaf@mellanox.com>

* [portstat] fix print table_as_json

Signed-off-by: Mykola Faryma <mykolaf@mellanox.com>

* Fix tx_ovr field

padmanarayana committed:
On sflow enable, start the docker since it is disabled by default (#773)

rvisnu committed:
[show] enhance 'show ip[v6] bgp summary' command (#754)

* [show] enhance 'show ip[v6] bgp summary' command
* changing ipaddr to ipaddress

daall committed:
[dropconfig] Clean-up error output for devices that do not support drop counters (#784)

Signed-off-by: Danny Allen <daall@microsoft.com>

cony7642 committed:
[config] Merge duplicate vrf configuration (#759)

Merge two vrf configuration to avoid user can't add or delete management VRF through CLI.

Co-authored-by: Minkang-Tsai <58835052+Minkang-Tsai@users.noreply.github.com>

antony-rheneus committed:
[sonic_installer] Adding support for ARM Uboot firmware (#769)

* [sonic_installer] Adding support for ARM by introducing bootloader type
grub vs uboot

For uboot fw_setenv/fw_printenv commands are utilized here
fw_env.cfg config needs to be updated per platform

Signed-off-by: Antony Rheneus <arheneus@marvell.com>

rvisnu committed:
ipaddr module to ipaddress for show command (#786)

jleveque committed:
[fdbshow][nbrshow] Print interface OID in lieu of name if there is no OID->interface name mapping (#789)

In release branches prior to 201911, FDB or ARP learnt on VLAN interface LAG port will cause nothing to appear in the "show arp" or "show mac" output because there existed no OID->interface name never had the mappings for LAG interface.

This issue has since been fixed in master (and 201911). However, making this change in master is not harmful and will prevent a regression from causing no output to be displayed for the interface name in the future.

Junchao-Mellanox committed:
Add command line support for thermal control  (#777)

Add a "show platform fanstatus" command to allow user fetch FAN status data.
Add a "show platform temperature" command to allow user fetch thermal status data.

olivier-singla committed:
kdump support (#729)

In the event of a kernel crash, we need to gather as much information as possible to understand and identify the root cause of the crash. Currently, the kernel does not provide much information, which make kernel crash investigation difficult and time consuming.

Fortunately, there is a way in the kernel to provide more information in the case of a kernel crash. kdump is a feature of the Linux kernel that creates crash dumps in the event of a kernel crash. This PR will add kernel kdump support. Please note that there is another PR in sonic-utilities which is also needed:
Azure/sonic-buildimage#3722

An extension to the CLI utilities config and show is provided to configure and manage kdump:

view kdump status (enabled/disabled, active, configuration, stored crash files)
enable / disable kdump functionality
configure kdump (how many kernel crash logs can be saved, memory
allocated for capture kernel)
view kernel crash logs
There is a design document which describes this kdump implementation:
Azure/SONiC#510

jleveque committed:
[acl-loader] Use V6 ethertype for IPv6 ACL rule (#788)

If ACL table name contains the substring "v6", set the EtherType of the rule to V6, otherwise set to V4.

daall committed:
[dropconfig] Move prefix-handling from CLI to orchagent (#785)

- Remove prefix-trimming from drop reason capability query
- Remove prefix-trimming from create/add safety checks

Signed-off-by: Danny Allen <daall@microsoft.com>

yxieca committed:
[neighbor advertiser] remove http endpoint access (#792)

* [neighbor advertiser] remove http endpoint access

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

* Remove obsolete comment

lguohan committed:
Changes in swss-utilities submodule to support NAT feature. (#645)

Signed-off-by: akhilesh.samineni@broadcom.com

lguohan committed:
[showtech]: dump a few more DB in generate dump (#795)

dump a few more DB, config db, state db, flex counter db in generate dump.
also do not zip some bgp table since the whole tar file is gziped.

heidinet2007 committed:
Add document for warm restart use case (#720)

* Add document for warm restart use case

* Address 1st round review comments

* Address review comments on warm restart section

* Remove "unplanned" sentence, due to the delay of dynamic capability support

dzhangalibaba committed:
[MultiDB] refactor docker warm reboot codes (#799)

borys-karpliak committed:
Fix 'show platform eeprom' on virtual switch (#794)

add kvm to platforms which do not support syseeprom db and check for the 'read_syseeprom_db' method availibilty to avoid same issue on other platforms

yozhao101 committed:
[show] Add subcommand to display the status of auto-restart feature for each container (#798)

Since we introduced the auto-restart feature for each container, we need add a show subcommand to display the current status of auto-restart feature for all containers or a specific container.

- How I did it
We define a function named autorestart to show the status of this features. This function will accept an option parameter which is the container name. If this parameter is not specified, this function will by default show the status of all containers. Otherwise it will show the status of specific container.

Signed-off-by: Yong Zhao <yozhao@microsoft.com>

yozhao101 committed:
[config] Add subcommand to configure the status of auto-restart feature for each container (#801)

yozhao101 committed:
[Command Reference] Document CLI commands related to container auto-restart feature (#802)

yozhao101 committed:
[Command Reference] Correct a typo and fix link jump in the section container auto-restart feature. (#808)

* [Command-reference.md] Correct a typo.

Signed-off-by: Yong Zhao <yozhao@microsoft.com>

* [Command-Reference.md] Fix the link jump.

Signed-off-by: Yong Zhao <yozhao@microsoft.com>

* [Command-Reference.md] Fix the link jump.

Signed-off-by: Yong Zhao <yozhao@microsoft.com>

* [Command-Reference.md] Fix the link jump.

Signed-off-by: Yong Zhao <yozhao@microsoft.com>

rajendra-dendukuri committed:
ZTP CLI commands (#599)

Implemented following commands to use Zero Touch Provisioning

show ztp status
config ztp enable
config ztp disable
config ztp run

Signed-off-by: Rajendra Dendukuri <rajendra.dendukuri@broadcom.com>
Co-authored-by: Akhilesh Samineni <47657796+AkhileshSamineni@users.noreply.github.com>
Co-authored-by: rlhui <48738894+rlhui@users.noreply.github.com>

yozhao101 committed:
[config/main.py] Modify reload() function to load configuration from init_cfg.json implicitly (#812)

* [config/main.py] Modify the load() and reload() functions to load config
from config_db.json and init_cfg.json.

Signed-off-by: Yong Zhao <yozhao@microsoft.com>

* [config/main.py] Undo the changes which load the configuration from
init_cfg.json for load() function and define a
constant string for the path of init_cfg.json.

Signed-off-by: Yong Zhao <yozhao@microsoft.com>

* [config/main.py] Correct a typo.

Signed-off-by: Yong Zhao <yozhao@microsoft.com>

* [config/main.py] Added an else statement in line 551 to decide whether
the init_cfg.json exsits or not in reload function.

Signed-off-by: Yong Zhao <yozhao@microsoft.com>

* [config/main.py] Correct a typo error.

Signed-off-by: Yong Zhao <yozhao@microsoft.com>

* [config/main.py] Change the loading order and we should first load
init_cfg.json and then config_db.json.

Signed-off-by: Yong Zhao <yozhao@microsoft.com>

* [config/main.py] Use constant string to represent the path of
init_cfg.json.

Signed-off-by: Yong Zhao <yozhao@microsoft.com>

nazariig committed:
[fwutil]: Command-line utility for interacting with platform components (#772)

* [fwutil]: initial version.

Signed-off-by: Nazarii Hnydyn <nazariig@mellanox.com>

* [fwutil]: Fix UI: enable progressbar render finalizer.

Signed-off-by: Nazarii Hnydyn <nazariig@mellanox.com>

* [fwutil]: integrate utility with SONiC CLI.

Signed-off-by: Nazarii Hnydyn <nazariig@mellanox.com>

* [fwutil]: update CLI command reference documentation.

Signed-off-by: Nazarii Hnydyn <nazariig@mellanox.com>

* [fwutil]: Revisit CLI architecture: avoid direct imports.

Signed-off-by: Nazarii Hnydyn <nazariig@mellanox.com>

* [fwutil]: Fix review comments: refactor CLI command reference.

Signed-off-by: Nazarii Hnydyn <nazariig@mellanox.com>

* [fwutil]: Fix review comments: update CLI documentation.

Signed-off-by: Nazarii Hnydyn <nazariig@mellanox.com>

okuzovych committed:
[config] Add 'config interface mtu' command (#793)

abdosi committed:
Added check if service existis before start/stop/restart of service. (#815)

This is possible if some of docker are not built then corresponding
service file will not be present

AkhileshSamineni committed:
Fix for dhcp_relay deletion on a VLAN (#813)

Signed-off-by: Akhilesh Samineni <akhilesh.samineni@broadcom.com>

jleveque committed:
[Command-Reference.md] Unify Usage statments and Examples (including sample prompts) (#816)

AkhileshSamineni committed:
Multi-Db changes for NAT feature. (#818)

yxieca committed:
[reboot] make sure the reboot happens even if platform reboot failed (#819)

* [reboot] make sure the reboot happens even if platform reboot failed

Platform reboot tool could fail if the platform driver didn't load
properly. In this case, a reboot is required to recover the device.
We need to make sure the reboot tool is robust.

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

* Rewording

* add log message

shine4chen committed:
add support for MCLAG (#453)

* add support for MCLAG

Signed-off-by: shine.chen <shine.chen@nephosinc.com>

* add warm-reboot support for ICCPd

Signed-off-by: shine.chen <shine.chen@nephosinc.com>

* ensure iccpd is there before stop iccpd

Signed-off-by: shine.chen <shine.chen@mediatek.com>

* fix service function

* remove unused comment

* refactor code according to feature management mechanism

Signed-off-by: shine.chen <shine.chen@mediatek.com>

andriymoroz-mlnx committed:
Merge branch 'master' into pfcwd_connect
